### PR TITLE
Added noble callback function signatures

### PR DESCRIPTION
**Pull request by LukeL99**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
Start Scanning
https://github.com/sandeepmistry/noble#start-scanning
Stop Scanning
https://github.com/sandeepmistry/noble/blob/master/lib/noble.js#L129
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/noble/index.d.ts [hunk 1]
```diff
@@ -9,9 +12,13 @@
 import events = require("events");
 
 export declare function startScanning(): void;
+export declare function startScanning(callback: Function): void;
```
Comment (mhegazy) [inline types/noble/index.d.ts]: do not use `Function` use `(....): void`

File: types/noble/index.d.ts [hunk 2]
```diff
@@ -9,9 +12,13 @@
...
 export declare function startScanning(serviceUUIDs: string[]): void;
+export declare function startScanning(serviceUUIDs: string[], callback: Function): void;
 export declare function startScanning(serviceUUIDs: string[], allowDuplicates: boolean): void;
+export declare function startScanning(serviceUUIDs: string[], allowDuplicates: boolean, callback: Function): void;
 export declare function stopScanning(): void;
+export declare function stopScanning(callback: Function): void;
```
Comment (mhegazy) [inline types/noble/index.d.ts]: combine with previous signature, 
```ts
export declare function stopScanning(callback?: Function)
```

**Review by mhegazy — CHANGES_REQUESTED** [review]

File: types/noble/index.d.ts (hunk 1 above)
Comment (LukeL99) [inline types/noble/index.d.ts]: Not exactly sure what you mean here. Would this be better?

`export declare function startScanning(callback?: (error?: Error) => void): void;`

File: types/noble/index.d.ts (hunk 2 above)
Comment (LukeL99) [inline types/noble/index.d.ts]: :+1: I thought I read that was a bad practice in the FAQ, but I realized that was only for getters/setters.

**Comment by swook** [comment]
Hello @LukeL99  please merge with branch master again to have this PR ready for merging.

**Comment by LukeL99** [comment]
Sorry @swook & @yuit , was traveling. Thanks for merging!

:+1:

**Pull request merged**